prefix-hex/src/data.rs

```diff
@@ -54,8 +54,16 @@ where
     }
 }
 
-impl ToHexPrefixed for &[u8] {
-    fn to_hex_prefixed(self) -> String {
-        format!("0x{}", hex::encode(self))
-    }
+macro_rules! impl_for_as_ref_type {
```

Comment:
Would it not be sufficient to impl it for `AsRef<[u8]>` ?

Reply:
That is what I was mentioning earlier that might clash with the `primitive_types` stuff in the future, that's why the compiler won't let me do it. Or is there another way?

Reply:
Ah alright, I guess it's fine then